src/hotspot/share/opto/macroArrayCopy.cpp

```diff
@@ -818,9 +825,18 @@ Node* PhaseMacroExpand::generate_arraycopy(ArrayCopyNode *ac, AllocateArrayNode*
     // Do not let stores that initialize this object be reordered with
     // a subsequent store that would make this object accessible by
     // other threads.
-    insert_mem_bar(ctrl, &out_mem, Op_MemBarStoreStore);
+    assert(ac->_dest_type == TypeOopPtr::BOTTOM, "non escaping destination shouldn't have narrow slice");
+    insert_mem_bar(ctrl, &out_mem, Op_MemBarStoreStore, Compile::AliasIdxBot);
   } else {
-    insert_mem_bar(ctrl, &out_mem, Op_MemBarCPUOrder);
+    int alias_idx = Compile::AliasIdxBot;
```

Comment:
Minor thing, `alias_idx` is already defined in the method with a different type. Would it make sense to use a different name here?

Earlier definition:
```cpp
uint alias_idx = C->get_alias_index(adr_type);
```

Reply:
Good catch. I renamed `alias_idx`.